src/components/AmountFilter.jsx

```diff
@@ -19,14 +22,27 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export default function AmountFilter() {
+export default function AmountFilter(props) {
   const classes = useStyles();
+  const { amountFilterVals } = props;
   const [anchorEl, setAnchorEl] = useState(null);
-  const [values, setValues] = useState({ min: 0, max: 0 });
+
+  const location = useLocation();
+  const history = useHistory();
 
   const handleClick = (event) => setAnchorEl(event.currentTarget);
-  const handleChange = (index, val) =>
-    setValues({ ...values, [index]: parseInt(val, 10) || '' });
+  const error =
+    amountFilterVals.maxAmount > 0 &&
+    amountFilterVals.maxAmount <= amountFilterVals.minAmount;
+
+  const setMinMax = (index, val) => {
+    history.push({
+      search: queryString.stringify({
+        ...queryString.parse(location.search),
+        [index]: val,
+      }),
+    });
+  };
```

Comment:
It seems a bit strange for query params to be set here, but read in Scholarships.jsx to then later set the props here. You _could_ technically just read/set the query params here instead of accepting properties. But, since the value is parsed, validated and used in Scholarships.jsx, perhaps it makes more sense to pass to AmountFilter `setMax`/`onMaxChange` and `setMin`/`onMinChange` functions instead? It kinda helps keeps query param specific business in one place since everywhere else just needs the parsed value.

Reply:
that makes sense. I moved that logic to Scholarships.jsx and pass it to AmountFitler as one function instead of 2, and like the function above, it uses an index to determine if it should set `minAmount` or `maxAmount`.